frontend/packages/process-editor/src/hooks/useBpmnEditor.ts

```diff
@@ -7,53 +7,56 @@ import type { TaskEvent } from '../types/TaskEvent';
 import type { SelectionChangedEvent } from '../types/SelectionChangeEvent';
 import { getBpmnEditorDetailsFromBusinessObject } from '../utils/bpmnObjectBuilders';
 import { useStudioRecommendedNextActionContext } from '@studio/components';
+import type Modeler from 'bpmn-js/lib/Modeler';
 
 // Wrapper around bpmn-js to Reactify it
 
 export type UseBpmnEditorResult = (div: HTMLDivElement) => void;
 
 export const useBpmnEditor = (): UseBpmnEditorResult => {
-  const { getUpdatedXml, bpmnXml, modelerRef, setBpmnDetails } = useBpmnContext();
+  const { getUpdatedXml, setBpmnDetails } = useBpmnContext();
   const { metadataFormRef, resetForm } = useBpmnConfigPanelFormContext();
   const { addAction } = useStudioRecommendedNextActionContext();
 
   const { saveBpmn, onProcessTaskAdd, onProcessTaskRemove } = useBpmnApiContext();
 
-  const handleCommandStackChanged = async () => {
-    saveBpmn(await getUpdatedXml(), metadataFormRef.current || null);
+  const handleCommandStackChanged = useCallback(async () => {
+    const xml = await getUpdatedXml();
+    saveBpmn(xml, metadataFormRef.current || null);
```

Comment:
Do we need to add `|| null` here? 🫣

Reply:
Not sure. Again, I have mainly focused on fixing state management patterns, keeping the actual event listener functions untouched, to be sure I don't mess up anything. Let's postpone this to a future cleanup session.